micronaut-mcp-server-java-sdk/src/main/java/io/micronaut/mcp/server/registry/JsonMapperTypedCallToRequestArgumentBinder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.mcp.server.registry;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.convert.ArgumentConversionContext;
+import io.micronaut.core.convert.ConversionError;
+import io.micronaut.core.type.Argument;
+import io.micronaut.json.JsonMapper;
+import io.modelcontextprotocol.spec.McpSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * {@link TypedCallToRequestArgumentBinder} which uses {@link JsonMapper} to bind to a specific type.
+ * @param <T> The argument type
+ */
+@Internal
+class JsonMapperTypedCallToRequestArgumentBinder<T> implements TypedCallToRequestArgumentBinder<T> {
```

Comment:
can be `final`

Reply:
resolved via https://github.com/micronaut-projects/micronaut-mcp/pull/61/commits/29e2d86d4669a8f9afc4b7d39cae53691537f46a